src/CommunityToolkit/Xamarin.CommunityToolkit/Effects/Touch/PlatformTouchEffect.uwp.cs

```diff
@@ -0,0 +1,205 @@
+﻿using Windows.UI.Xaml.Input;
+using Xamarin.Forms;
+using Xamarin.Forms.Internals;
+using Xamarin.Forms.Platform.UWP;
+using Windows.UI.Xaml.Media.Animation;
+using System;
+using System.Collections.Generic;
+using Xamarin.CommunityToolkit.UWP.Effects;
+using Xamarin.CommunityToolkit.Effects;
+
+[assembly: ExportEffect(typeof(PlatformTouchEffect), nameof(TouchEffect))]
+
+namespace Xamarin.CommunityToolkit.UWP.Effects
+{
+	public class PlatformTouchEffect : PlatformEffect
+	{
+		TouchEffect effect;
+
+		bool isPressed;
+
+		bool isIntentionalCaptureLoss;
+
+		Storyboard pointerDownStoryboard;
+
+		Storyboard pointerUpStoryboard;
+
+		protected override void OnAttached()
+		{
+			effect = TouchEffect.PickFrom(Element);
+			if (effect?.IsDisabled ?? true)
+				return;
+
+			effect.Element = Element as VisualElement;
```

Comment:
@AndreiMisiukevich Would not be better to throw an exception if the `Element` isn't a `VisualElement` (VE), that way is very clear to the user that the type should be a VE. Just like we use to cast the `BindableObject` on `BindableProperties`
```suggestion
			effect.Element = (VisualElement)Element;
```
This also applies to other platforms.

Reply:
Done